### PR TITLE
Logger status

### PR DESCRIPTION
**Pull request by idan-orbs**

### CODE REVIEW

File: logging/src/main.ts [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
 import { request, ClientRequest, RequestOptions, IncomingMessage } from "http";
+import {writeStatusToDisk} from "./status";
 
 const app: Express = express();
 const port: number = 80;
 
+const serviceLaunchTime = Math.round(new Date().getTime() / 1000);
+const statusFilePath = process.env.STATUS_FILE_PATH || "/opt/orbs/status/status.json";
+
+let error = '';
```
Comment (Luke-Rogerson) [inline logging/src/main.ts:11]: I asked ChatGPT for advice about how we can avoid this "global" error variable and it suggested this:

```typescript
import express, { Express, NextFunction, Request, Response } from "express";
import { request, ClientRequest, RequestOptions, IncomingMessage } from "http";
import { writeStatusToDisk } from "./status";

const app: Express = express();
const port: number = 80;

const serviceLaunchTime = Math.round(new Date().getTime() / 1000);
const statusFilePath = process.env.STATUS_FILE_PATH || "/opt/orbs/status/status.json";

setInterval(() => {
    writeStatusToDisk(statusFilePath, serviceLaunchTime);
}, 5 * 60 * 1000);

const validNameRegex = /^[a-zA-Z0-9][a-zA-Z0-9_.-]*$/;

app.use((_: Request, res: Response, next: NextFunction) => {
    res.setHeader("Content-Disposition", "inline");
    res.setHeader("Content-Type", "text/plain; charset=utf-8");
    next();
});

const decodeDockerLogs = (data: Buffer): string => {
    // ... [no change in this function]
};

app.get("/service/:name/log", (req: Request, res: Response, next: NextFunction) => {
    const containerName: string = req.params.name;

    if (!validNameRegex.test(containerName)) {
        return next(new Error("Invalid container name"));
    }

    const options: RequestOptions = {
        // ... [no change in this part]
    };

    const clientRequest: ClientRequest = request(options, (resp: IncomingMessage) => {
        // ... [no change in this function]
    });

    clientRequest.on("error", (e: Error) => {
        next(e); // Pass the error to the error handling middleware
    });

    clientRequest.end();
});

// Dedicated error handling middleware
app.use((err: Error, req: Request, res: Response, next: NextFunction) => {
    console.error(err.message);
    writeStatusToDisk(statusFilePath, serviceLaunchTime, err.message); // Log the error
    res.status(500).send("An unexpected error occurred. Try again later");
});

app.listen(port, () => {
    console.log(`Logging service listening at http://localhost:${port}`);
});
```

Comment (Luke-Rogerson) [reply logging/src/main.ts:11]: What do you think? I do like that it centralises the whole error handling logic

File: logging/src/main.ts [hunk 2]
```diff
@@ -1,9 +1,21 @@
 import express, { Express, NextFunction, Request, Response } from "express";
 import { request, ClientRequest, RequestOptions, IncomingMessage } from "http";
+import {writeStatusToDisk} from "./status";
```
Comment (Luke-Rogerson) [inline logging/src/main.ts:3]: Can I be annoying and ask you to run Prettier on this file? Formatting is quite different in places

File: logging/src/status.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    console.log(`Wrote status JSON to ${filePath} (${content.length} bytes).`);
+}
+
+export function getCurrentVersion() {
+    try {
+        return fs.readFileSync('/app/.version').toString().trim();
+    } catch (err: any) {
+        console.error(`Could not find version: ${err.message}`);
+    }
+    return '';
```
Comment (Luke-Rogerson) [inline logging/src/status.ts:42]: Maybe just return something like "Version not found" for clarity?

File: setup/scripts/health-check.sh [hunk 4]
```diff
@@ -1,14 +1,34 @@
 #!/bin/bash
 
-echo -e "${BLUE}Performing a health check...${NC}\n"
+check_services() {
```
Comment (Luke-Rogerson) [inline setup/scripts/health-check.sh:3]: Could also look at using `docker-compose -d --wait` in the future - https://maciejwalkowiak.com/blog/docker-compose-waiting-containers-ready/

File: logging/src/main.ts (hunk 1 above)
Comment (idan-orbs) [reply logging/src/main.ts:11]: So how does it work? `app.use` is handling the errors, and `next` is in charge of surfacing them?

File: setup/scripts/health-check.sh (hunk 4 above)
Comment (idan-orbs) [reply setup/scripts/health-check.sh:3]: Nice, wasn't familiar.
I tried it now locally but it doesn't work. Probably b/c services are stuck on "starting" and not "healthy".
Creating a follow-up ticket

**Pull request merged**